packages/discovery-provider/src/models/users/email.py

```diff
@@ -8,53 +8,69 @@
+    Table storing encrypted emails for users.
     """
 
     __tablename__ = "encrypted_emails"
 
     id = Column(Integer, primary_key=True)
     email_owner_user_id = Column(Integer, nullable=False)
-    primary_user_id = Column(Integer, nullable=False)
     encrypted_email = Column(Text, nullable=False)
     created_at = Column(DateTime(timezone=True), server_default=func.now())
     updated_at = Column(
         DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
     )
 
-
-class EmailEncryptionKey(Base, RepresentableMixin):
-    """Model class for the email_encryption_keys table.
-
-    Table storing encryption keys for users with primary access to manage encrypted emails.
-    """
-
-    __tablename__ = "email_encryption_keys"
-
-    id = Column(Integer, primary_key=True)
-    primary_user_id = Column(Integer, nullable=False, unique=True)
-    encrypted_key = Column(Text, nullable=False)
-    created_at = Column(DateTime(timezone=True), server_default=func.now())
-    updated_at = Column(
-        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
+    __table_args__ = (
+        UniqueConstraint("email_owner_user_id", name="uq_encrypted_email_owner"),
     )
 
+    def to_dict(self):
+        """Convert model to dictionary."""
+        return {
+            "id": self.id,
+            "email_owner_user_id": self.email_owner_user_id,
+            "encrypted_email": self.encrypted_email,
+            "created_at": self.created_at.isoformat(),
+            "updated_at": self.updated_at.isoformat(),
+        }
+
 
-class EmailAccessKey(Base, RepresentableMixin):
-    """Model class for the email_access_keys table.
+class EmailAccess(Base, RepresentableMixin):
+    """Model class for the email_access table.
 
-    Table storing encrypted keys for users with delegated access to view emails.
+    Table storing access grants and encrypted keys for email access.
     """
 
-    __tablename__ = "email_access_keys"
+    __tablename__ = "email_access"
 
     id = Column(Integer, primary_key=True)
-    primary_user_id = Column(Integer, nullable=False)
-    delegated_user_id = Column(Integer, nullable=False)
+    email_owner_user_id = Column(Integer, nullable=False)
+    receiving_user_id = Column(Integer, nullable=False)
+    grantor_user_id = Column(Integer, nullable=False)
     encrypted_key = Column(Text, nullable=False)
     created_at = Column(DateTime(timezone=True), server_default=func.now())
     updated_at = Column(
         DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
     )
 
-    __table_args__ = (UniqueConstraint("primary_user_id", "delegated_user_id"),)
+    __table_args__ = (
+        UniqueConstraint(
+            "email_owner_user_id",
+            "receiving_user_id",
+            "grantor_user_id",
+            name="uq_email_access",
+        ),
+    )
+
+    def to_dict(self):
```

Comment:
we do have some helpers for this fyi, but i like what you did here in the class. it's cleaner
https://github.com/AudiusProject/audius-protocol/blob/874ed4dd30cec590e3154b88d45d954e9ef93652/packages/discovery-provider/src/utils/helpers.py#L98